cognite/neat/_session/_read.py

```diff
@@ -167,6 +168,27 @@ def __call__(self, io: Any, type: str, primary_key: str) -> None:
         self._state.instances.store.write(extractor)
 
 
+@intercept_session_exceptions
+class XMLReadAPI(BaseReadAPI):
```

Comment:
Suggestion also implement it similar to what we did for RDF:

https://github.com/cognitedata/neat/blob/51a863bce49e014ebb7d1f1a7339d36a74abe528/cognite/neat/_session/_read.py#L218C1-L244C84